proposals/js-promise-integration/Overview.md

```diff
@@ -177,4 +261,34 @@ A function is suspendable if it was
 * returned by `returnPromiseOnSuspend`,
 * or generated by [creating a host function](https://webassembly.github.io/spec/js-api/index.html#create-a-host-function) for a suspendable function
 
-Importantly, functions written in JavaScript are *not* suspendable, conforming to feedback from members of [TC39](https://tc39.es/), and host functions (except for the few listed above) are *not* suspendable, conforming to feedback from engine maintainers.
\ No newline at end of file
+Importantly, functions written in JavaScript are *not* suspendable, conforming to feedback from members of [TC39](https://tc39.es/), and host functions (except for the few listed above) are *not* suspendable, conforming to feedback from engine maintainers.
+
+## Frequently Asked Questions
+
+1. **What is the purpose of the `Suspender` object?**
+
+   The `Suspender` object is used to connect a `Promise` returning import with a `Promise` returning export. Without this explicit connection, it becomes problematic especially when constructing so-called chains of modules: where one module calls into the exports of another.
+
+1. **Why do we try to prevent JavaScript programs from using this API?**
+
+   JavaScript already has a way of managing computations that can suspend. This is semantically connected to JavaScript `Promise` objects and the `async` function syntax. However, a more important reason is that it is important, in the context of JavaScript, that we do not introduce language features that can affect the behavior of existing programs.
+
+1. **Why does the API only apply t `WebAssembly.Function` values, and not `Function`s?**
+
+   The core interface elements in this API refer to the use of `WebAssembly.Function` entities rather than plain `Function` entities. There are some potential questions and issues about this choice:
+
+    * Compared to regular imports, it is arguably less ergonomic to manually mark the types of a function when importing a JavaScript function as a wrapped import. This is not needed for normal imports, but we are requiring it for wrapped imports.
+    * A wrapped import function is not callable from JavaScript. The role of a wrapped import is fundamentally to signal intentions to the process of instantiating modules. It is arguable that using wrapper functions in this way may not be the best architectural approach to addressing this signaling.
+
```

Comment:
(That is, I assume the answer is something like "explicitly using `WebAssembly.Function` forces the user to annotate the type, which makes things easier in the engine `[somehow]`", but I don't see that here? I also don't know what that "`somehow`" is :smile: )

Reply:
It makes a pretty substantial difference, especially for the 'static' version of the API. That is because the wrapper functions we generate have to both call their wrappee functions (is that a word?) and change their arguments (add/subtract a suspender).
In addition, there is the issue of creating a Function that is not callable from JavaScript. Seems like an oxymoron.

Reply:
Ok, I guess this gets into VM details that I'm not familiar enough with.